samples/bluetooth/iso_broadcast/src/main.c

```diff
@@ -197,14 +212,87 @@ int main(void)
 			net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
 			sys_put_le32(iso_send_count, iso_data);
 			net_buf_add_mem(buf, iso_data, sizeof(iso_data));
-			ret = bt_iso_chan_send(&bis_iso_chan[chan], buf, seq_num);
+
+			/* Ensure that first ISO data on all streams are given with sequence number
+			 * 0, so that the (Zephyr) Controller uses the same payload counter for
+			 * a given sequence number on all streams at an SDU interval.
+			 * Read Tx Sync interface returns error if no ISO data was given previously.
+			 */
+			ret = bt_iso_chan_get_tx_sync(&bis_iso_chan[chan], &tx_info);
+			if (ret) {
+				printk("Failed to get tx sync before send (%d)\n", ret);
+
+				/* Use sequence number 0, to synchronize current stream with the
+				 * current payload counter in the ISO group of streams.
+				 */
+				sn = 0U;
+
+			} else {
+				sn = seq_num;
+			}
+
+			/* Send ISO data with timestamp, use sequence number 0 for first ISO data on
+			 * any stream.
+			 */
+			if (sn) {
+				ret = bt_iso_chan_send_ts(&bis_iso_chan[chan], buf, sn, ts);
+			} else {
+				ret = bt_iso_chan_send(&bis_iso_chan[chan], buf, sn);
+			}
```

Comment:
Checking for `sn == 0` won't work when it wraps.

Reply:
true, what do you suggest to use for checking if it is the first SDU being sent?